### PR TITLE
[FAW] FAW embedding use LRU as eviction strategy intialized with dataset stats

### PR DESCRIPTION
**Pull request by CsRic**

### CODE REVIEW

**Comment by CsRic** [comment]
lfu strategy has been tested on real dlrm training case . no bug now.

File: colossalai/nn/parallel/layers/cache_embedding/cache_mgr.py [hunk 1]
```diff
@@ -50,8 +50,10 @@ def __init__(self,
         if self._evict_strategy == EvictionStrategy.LFU:
             # cpu_row_idx -> frequency, freq of the cpu rows.
             # evict the minimal freq value row in cuda cache.
+            
+            # the last element match to masked cache_idx, and should ALWAYS have MAX + 1 value.
```
Comment (feifeibear) [inline colossalai/nn/parallel/layers/cache_embedding/cache_mgr.py:54]: 这句comments用中文描述一下？
语法也不太对，the last element matches

**Review by feifeibear — APPROVED** [review]

File: colossalai/nn/parallel/layers/cache_embedding/cache_mgr.py [hunk 2]
```diff
@@ -298,26 +303,24 @@ def _prepare_rows_on_cuda(self, cpu_row_idxs: torch.Tensor) -> None:
...
                     evict_gpu_row_idxs = self._find_evict_gpu_idxs(evict_num)
                     self.cached_idx_map.index_copy_(0, invalid_idxs, backup_idxs)
                     
                 elif self._evict_strategy == EvictionStrategy.LFU:
-                    # another mask method.
-                    # set freq_cnter[invalid_idxs] to max
-                    # so those idxs will be sorted to end, therefore not being chosen as victim
-                    backup_cnter = self.freq_cnter[invalid_idxs].clone()
-                    self.freq_cnter.index_fill_(0, invalid_idxs, torch.max(self.freq_cnter) + 1) # or can we use a confident max value? 
+                    self.freq_cnter[-1] = torch.max(self.freq_cnter[:-1]) + 1
```
Comment (feifeibear) [inline colossalai/nn/parallel/layers/cache_embedding/cache_mgr.py:318]: 直接把它设成最大int呗？这样每次减少一个max操作？

Comment (CsRic) [reply colossalai/nn/parallel/layers/cache_embedding/cache_mgr.py]: ok

**Pull request merged**